### PR TITLE
Fixes #122 - Added Kinto.createRemoteTransformer().

### LINKED ISSUE #122: Using Transformers with prototypal inheritance
The `class` and `extends` keywords only work in ES6, and it would be nice if developers can define their Transformer using just ES5. I *think* this code is equivalent to the ES6 code in http://kintojs.readthedocs.org/en/latest/api/#transformers and that we should add it to the documentation?

In ES5:
````js
function MyRemoteTransformer() {};

MyRemoteTransformer.prototype = Kinto.transformers.RemoteTransformer.prototype;
MyRemoteTransformer.prototype.encode = function(record) {
  record.title += "!";
  return record;
};

MyRemoteTransformer.prototype.decode = function(record) {
  record.title = record.title.slice(0, -1);
  return record;
}

const kinto = new Kinto({remote: "https://my.server.tld/v1"});
coll = kinto.collection("articles");
coll.use(new MyRemoteTransformer());
````

In ES6:
````js
import Kinto from "kinto";

function update(obj1, obj2) {
  return Object.assign({}, obj1, obj2);
}

class MyRemoteTransformer extends Kinto.transformers.RemoteTransformer {
  encode(record) {
    return update(record, {title: record.title + "!"});
  }

  decode(record) {
    return update(record, {title: record.title.slice(0, -1)});
  }
}

const kinto = new Kinto({remote: "https://my.server.tld/v1"});
coll = kinto.collection("articles");
coll.use(new MyRemoteTransformer());
````

### PR DESCRIPTION
**Pull request by n1k0**
refs #122

### Creating transformers in ES5

If your JavaScript environment doesn't suppport [ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) just yet, you can derive transformers in an ES5 fashion using `Kinto.createRemoteTransformer` static helper:

```js
const TitleCharTransformer = Kinto.createRemoteTransformer({
  constructor: function(char) {
    this.char = char;
  },

  encode: function(record) {
    return update(record, {title: record.title + this.char});
  },

  decode: function(record) {
    return update(record, {title: record.title.slice(0, -1)});
  }
});

coll.use(new TitleCharTransformer("!"));
coll.use(new TitleCharTransformer("?"));
```

> #### Notes
>
> Notice you can define a `constructor` method, though you don't need (and must not) call `super()` within it, unlike with ES6 classes.

### CODE REVIEW

**Comment by n1k0** [comment]
r=? @michielbdejong

File: src/index.js [hunk 1]
```diff
@@ -44,6 +44,27 @@ export default class Kinto {
   }
 
   /**
+   * Creates a remote transformer constructor, the ES5 way.
+   *
+   * @return {RemoteTransformer}
+   */
+  static createRemoteTransformer(proto) {
+    if (!proto || typeof proto !== "object")
+      throw new Error("Expected prototype object.");
```
Comment (ametaireau) [inline src/index.js]: I though it was a good practice to always enclose iffed instructions inside a block `{` and `}`.

**Comment by ametaireau** [comment]
Looks good to me (with nitpick)

File: src/index.js (hunk 1 above)
Comment (n1k0) [inline src/index.js]: Depends on your coding style; Gecko (and Mozilla)'s one states that for single line statements braces can be omitted. If we're planning on removing this rule, we'll need to update a bunch of files elsewhere; if that's the case, it's probably safe to spawn another issue.

**Comment by n1k0** [comment]
Waiting for @michielbdejong final r+, ideally with checking that everything works fine on his end :)

File: src/index.js (hunk 1 above)
Comment (ametaireau) [inline src/index.js]: They actually state that you should always use them:

> Always brace controlled statements, even a single-line consequent of an if else else. This is redundant typically, but it avoids dangling else bugs, so it's safer at scale than fine-tuning.
> — [Developer Coding Style guide](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#Control_Structures)

Comment (n1k0) [inline src/index.js]: Bah, this must have changed recently, I remember we conformed to it for
Loop.

Still, could you please find a distinct issue about this?

Nicolas Perriault
Le 31 août 2015 7:31 PM, "Alexis Metaireau" <notifications@github.com> a
écrit :

Comment (n1k0) [inline src/index.js]: s/find/file (cannot edit comment from mobile)

**Comment by michielbdejong** [comment]
Will check today!

**Comment by michielbdejong** [comment]
The `encode` and `decode` methods of my transformer return a promise (can't do crypto stuff synchronously of course), and this seems to work. But the example transformer returns a value synchronously. Do both these approaches work? If so, maybe worth a note in the docs about that?

**Comment by n1k0** [comment]
> Do both these approaches work? 

Yes.

> If so, maybe worth a note in the docs about that?

Well, both approaches are documented:

- sync http://kintojs.readthedocs.org/en/latest/api/#remote-transformers
- async http://kintojs.readthedocs.org/en/latest/api/#async-transformers

**Comment by n1k0** [comment]
I can probably add a specific note about the accepted return types though, will update.

**Comment by michielbdejong** [comment]
Oh, great, I hadn't scrolled down to the async section, but now I do remember having seen it a while ago. Thanks a lot! r+

**Pull request merged**